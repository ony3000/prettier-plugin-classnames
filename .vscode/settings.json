{
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "editor.formatOnSave": true,
  "eslint.workingDirectories": [{ "mode": "auto" }]
}
