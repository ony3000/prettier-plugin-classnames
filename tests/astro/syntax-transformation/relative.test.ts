import { thisPlugin, testSnapshotEach } from '../../adaptor';
import { baseOptions } from '../../settings';
import { fixtures } from './fixtures';

const options = {
  ...baseOptions,
  plugins: ['prettier-plugin-astro', thisPlugin],
  parser: 'astro',
  printWidth: 60,
  endingPosition: 'relative',
};

testSnapshotEach(fixtures, options);
