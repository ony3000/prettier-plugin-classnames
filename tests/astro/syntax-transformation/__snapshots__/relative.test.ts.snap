// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`'(1) Short enough class names do not c…' > expectation 1`] = `
"<!-- ------------------------------------------------------| printWidth=60 (in snapshot) -->
<div>
  <div>
    <div class="lorem ipsum dolor sit amet">
      <slot />
    </div>
  </div>
</div>
"
`;

exports[`'(2) Long enough class names cause syn…' > expectation 1`] = `
"<!-- ------------------------------------------------------| printWidth=60 (in snapshot) -->
<div>
  <div>
    <div
      class={\`lorem ipsum dolor sit amet consectetur adipiscing elit proin
        ex massa hendrerit eu posuere eu volutpat id neque
        pellentesque\`}
    >
      <slot />
    </div>
  </div>
</div>
"
`;

exports[`'(3) This transformation does not supp…' > expectation 1`] = `
"<!-- ------------------------------------------------------| printWidth=60 (in snapshot) -->
<div>
  <div>
    <div class={"lorem ipsum dolor sit amet"}>
      <slot />
    </div>
  </div>
</div>
"
`;
